### PR TITLE
Changes to Matrix Classes 

### PR DESCRIPTION
**Pull request by ghost**
I added this[int column, int row] to all the matrix classes. I found it useful in Matrix4, and I use it in my code.
Mirror of [this pull request](https://github.com/giawa/opengl4csharp/pull/51), but I had to redo it after deleting the branch on accident.

This is updated to work with [this pull request](https://github.com/giawa/opengl4csharp/pull/55).

### CODE REVIEW

File: OpenGL/Math/Matrix2.cs [hunk 1]
```diff
@@ -86,6 +86,37 @@ public static Matrix2 Identity
             }
         }
 
+        public float this[int column, int row]
+        {
+            get
+            {
+                if (row == 0)
```
Comment (giawa) [inline OpenGL/Math/Matrix2.cs:93]: Should either have something like `if (row < 0 || row > 1) throw new AgumentOutOfRangeException(nameof(row))` or have the Exception under an else block and only check for row == 0 and row == 1.

File: OpenGL/Math/Matrix3.cs [hunk 2]
```diff
@@ -98,6 +98,49 @@ public static Matrix3 Identity
             }
         }
 
+        public float this[int column, int row]
+        {
+            get
+            {
+                if (row == 0)
```
Comment (giawa) [inline OpenGL/Math/Matrix3.cs:105]: Should either have something like `if (row < 0 || row > 2) throw new AgumentOutOfRangeException(nameof(row)), etc

File: OpenGL/Math/Matrix4.cs [hunk 3]
```diff
@@ -94,6 +94,64 @@ public static Matrix4 Identity
             }
         }
 
+        public float this[int column, int row]
+        {
+            get
+            {
+                if (row == 0)
```
Comment (giawa) [inline OpenGL/Math/Matrix4.cs:101]: Should either have something like `if (row < 0 || row > 2) throw new AgumentOutOfRangeException(nameof(row)), etc

File: OpenGL/Math/Matrix2.cs [hunk 4]
```diff
@@ -86,6 +86,37 @@ public static Matrix2 Identity
...
+            }
+            set
+            {
+                if (column > 1 || column < 0 || row > 1 || row < 0)
+                    throw new ArgumentOutOfRangeException("column or row is out of the matrix bounds.");
+
+                if (row == 0)
+                {
+                    if (column == 0) row1.X = value;
+                    if (column == 1) row1.Y = value;
```
Comment (giawa) [inline OpenGL/Math/Matrix2.cs]: else

File: OpenGL/Math/Matrix2.cs [hunk 5]
```diff
@@ -86,6 +86,37 @@ public static Matrix2 Identity
...
+                }
+                if (row == 1)
```
Comment (giawa) [inline OpenGL/Math/Matrix2.cs]: else

File: OpenGL/Math/Matrix2.cs [hunk 6]
```diff
@@ -86,6 +86,37 @@ public static Matrix2 Identity
...
+                {
+                    if (column == 0) row2.X = value;
+                    if (column == 1) row2.Y = value;
```
Comment (giawa) [inline OpenGL/Math/Matrix2.cs]: else

File: OpenGL/Math/Matrix3.cs [hunk 7]
```diff
@@ -98,6 +98,49 @@ public static Matrix3 Identity
...
+                if (column > 2 || column < 0 || row > 2 || row < 0)
+                    throw new ArgumentOutOfRangeException("column or row is out of the matrix bounds.");
+
+                if (row == 0)
+                {
+                    if (column == 0) row1.X = value;
+                    if (column == 1) row1.Y = value;
+                    if (column == 2) row1.Z = value;
+                }
+                if (row == 1)
```
Comment (giawa) [inline OpenGL/Math/Matrix3.cs]: else

File: OpenGL/Math/Matrix3.cs [hunk 8]
```diff
@@ -98,6 +98,49 @@ public static Matrix3 Identity
...
+                {
+                    if (column == 0) row2.X = value;
+                    if (column == 1) row2.Y = value;
+                    if (column == 2) row2.Z = value;
+                }
+                if (row == 2)
```
Comment (giawa) [inline OpenGL/Math/Matrix3.cs]: else

File: OpenGL/Math/Matrix3.cs [hunk 9]
```diff
@@ -98,6 +98,49 @@ public static Matrix3 Identity
...
+            }
+            set
+            {
+                if (column > 2 || column < 0 || row > 2 || row < 0)
+                    throw new ArgumentOutOfRangeException("column or row is out of the matrix bounds.");
+
+                if (row == 0)
+                {
+                    if (column == 0) row1.X = value;
+                    if (column == 1) row1.Y = value;
```
Comment (giawa) [inline OpenGL/Math/Matrix3.cs]: else if (for this and similar lines)

File: OpenGL/Math/Matrix3.cs [hunk 10]
```diff
@@ -98,6 +98,49 @@ public static Matrix3 Identity
...
+                    if (column == 2) row1.Z = value;
```
Comment (giawa) [inline OpenGL/Math/Matrix3.cs]: else (for this and similar lines)

File: OpenGL/Math/Matrix4.cs [hunk 11]
```diff
@@ -94,6 +94,64 @@ public static Matrix4 Identity
...
+                    return row4.Get(column);
+                }
+            }
+            set
+            {
+                if (column > 3 || column < 0 || row > 3 || row < 0)
+                    throw new ArgumentOutOfRangeException("column or row is out of the matrix bounds.");
+
+                //Checking each column is needed, so as not to use Vector4[int], as it is not supported by System.Numerics
+                if (row == 0)
```
Comment (giawa) [inline OpenGL/Math/Matrix4.cs:125]: Same comments as above regarding else/else if

**Review by giawa — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
Okay, I will fix all of these problems.

**Comment by ghost** [comment]
All the problems should be fixed.

**Pull request merged**